fcitx5-fbterm.cpp

```diff
@@ -0,0 +1,365 @@
+/*
+* SPDX-FileCopyrightText: 2021 duzhaokun123 <duzhaokun2@outlook.com>
+*
+* SPDX-License-Identifier: GPL-3.0-or-later
+*
+*/
+
+#include "fcitx5-fbterm.h"
+#include "imapi.h"
+#include "keycode.h"
+#include "keymap.h"
+#include "utils.h"
+#include <cstring>
+#include <fcitx-gclient/fcitxgclient.h>
+#include <fcitx-utils/capabilityflags.h>
+#include <fcitx-utils/keysymgen.h>
+#include <fcitx-utils/fs.h>
+#include <gio/gio.h>
+#include <iconv.h>
+
+#define eprintf(args...) fprintf(stderr, args)
+#define getArg1(i) getArg3(argc, argv, i)
+#define clearWin(id) set_im_window(id, rect0)
+#define BUF_SIZE 8192
+#define WINID_IM 0
+#define WINID_ERROR 1
+
+using namespace std;
+
+static char raw_mode = 1;
+static FcitxGClient* client;
+static GMainLoop* main_loop;
+static bool active = false;
+static fcitx::KeyState state;
+static char textup[BUF_SIZE];
+static char textdown[BUF_SIZE];
+static char imname[BUF_SIZE];
+static iconv_t iconvW;
+static int cursorPos = -1;
+static uint auxUpWidth;
+ColorType fc = Black;
```

Comment:
for variable name like this, please use full name instead of abbreviation. e.g. foregroundColor.